libs/MeshKernel/include/MeshKernel/Utilities/LinearAlgebra.hpp

```diff
@@ -1,22 +1,467 @@
+    /// @return    True if the matrix is empty, false otherwise
+    template <typename T, int storage>
+    [[nodiscard]] inline static bool MatrixIsEmpty(Matrix<T, storage> const& matrix)
+    {
+        return matrix.size() == 0;
+    }
+
+    /// @brief         Resizes and fills a matrix
+    /// @tparam        T        The data type of the matrix elements
+    /// @tparam        storage  Matrix storage option
+    /// @param[in,out] matrix   The matrix
+    /// @param[in]     rows     The number of rows
+    /// @param[in]     cols     The number of columns
+    /// @param[in]     preserve Optional parameter that determines if the existing elements in
+    ///                         the matrix should be preserved or overwritten by the provided
+    ///                         or default (if not provided) fill value
+    /// @param[in]     fill_value The optional fill value (defaults to the template class constructor)
+    /// @return        True if the input matrix has been resized, false otherwise
+    template <typename T, int storage>
+    inline static bool ResizeAndFillMatrix(Matrix<T, storage>& matrix,
+                                           Eigen::Index rows,
+                                           Eigen::Index cols,
+                                           bool preserve = false,
+                                           T const& fill_value = {})
+    {
+        if (rows < 0 || cols < 0)
+        {
+            throw meshkernel::LinearAlgebraError(meshkernel::VariadicErrorMessage(
+                "Invalid dimensions: rows = {}, cols = {}. Dimensions must be strictly positive.",
+                rows,
+                cols));
+        }
+
+        Eigen::Index const rows_old = matrix.rows();
+        Eigen::Index const cols_old = matrix.cols();
+        bool const resize = (rows != rows_old) || (cols != cols_old);
+        if (!resize)
+        {
+            if (!preserve)
+            {
+                matrix.fill(fill_value);
+            }
+        }
+        else
+        {
+            if (!preserve)
+            {
+                matrix.resize(rows, cols);
+                matrix.fill(fill_value);
+            }
+            else
+            {
+                matrix.conservativeResize(rows, cols);
+                if (rows > rows_old)
+                {
+                    matrix.bottomRows(rows - rows_old).setConstant(fill_value);
+                }
+                if (cols > cols_old)
+                {
+                    matrix.topRightCorner(rows_old, cols - cols_old).setConstant(fill_value);
+                }
+            }
+        }
+        return resize;
+    }
+
+    /// @brief         Erases a range of rows from a matrix given the begin and end row indices
+    /// @tparam        T         Matrix data type
+    /// @tparam        storage   Matrix storage option
+    /// @param[in,out] matrix    The matrix
+    /// @param[in]     col_begin The begin row index
```

Comment:
the name of the parameters in the documentation is different from the actual parameters

Reply:
Fixed